demos/src/iot_demo_mqtt.c

```diff
@@ -416,6 +417,10 @@ static int _establishMqttConnection( bool awsIotMqttMode,
     connectInfo.cleanSession = true;
     connectInfo.keepAliveSeconds = KEEP_ALIVE_SECONDS;
     connectInfo.pWillInfo = &willInfo;
+    connectInfo.pUserName = pCredentials->pUserName;
+    connectInfo.userNameLength = ( uint16_t ) pCredentials->userNameSize;
+    connectInfo.pPassword = pCredentials->pPassword;
+    connectInfo.passwordLength = ( uint16_t ) pCredentials->passwordSize;
```

Comment:
If `demoArguments.securedConnection == true` in `iot_demo.c` (as is in CI), this will attempt to dereference a null pointer, as the network credential info pointer won't be set. `pCredentials != NULL` needs to be checked before executing these lines.

Reply:
Yes, good catch, I had missed that. Fix appears to have worked, per latest CI results.